src/librustc_session/lint/builtin.rs

```diff
@@ -532,6 +532,12 @@ declare_lint! {
     "unsafe operations in unsafe functions without an explicit unsafe block are deprecated",
 }
 
+declare_lint! {
+    pub UNUSED_ASM_ARGUMENTS,
+    Warn,
+    "inline asm arguments not used in the template string",
+}
```

Comment:
Do I need to add this to `HardwiredLints` below? I'm a bit unsure about this since not all builtin lints are included in it.

Reply:
It needs to be added to `HardwiredLints`, otherwise `allow(unused_asm_arguments)` won't work.